fastestimator/estimator.py

```diff
@@ -215,10 +215,18 @@ def _warmup(self, eager: bool = True) -> None:
                         if idx > 0:  # ignore TrainEssential and EvalEssential's inputs for unmet requirement checking
                             trace_input_keys.update(trace.inputs)
                         trace_output_keys.update(trace.get_outputs(ds_ids=ds_ids))
+                    # Mode checking
+                    if mode == "train":
```

Comment:
rather than duplicating this mode check code let's modify system to have:

`
@ property
def steps_per_epoch(self):
  if self.mode == 'train':
     return self.train_steps_per_epoch
  ...
`